test/test_bson.py

```diff
@@ -650,6 +650,89 @@ def test_utf8(self):
             z = {iso8859_bytes: "hello"}
             self.assertRaises(InvalidStringData, BSON.encode, z)
 
+    # Verify that python and bson have the same understanding of
+    # legal utf-8 if the first byte is 0xf4 (244)
+    @staticmethod
+    def _py_is_legal_utf8(x):
+        try:
+            x.decode('utf-8')
+            return True
+        except UnicodeDecodeError:
+            return False
+
+    @staticmethod
+    def _bson_is_legal_utf8(x):
+        try:
+            BSON.encode({'x': x})
+            return True
+        except InvalidStringData:
+            return False
+
+    @unittest.skipIf(PY3, "python3 has strong separation between bytes/unicode")
+    def test_legal_utf8_full_coverage(self):
```

Comment:
Since this test takes so long we will probably separate it out into its own file and only run it once for each python 2.x version. But don't worry about this for now.